astropy/visualization/mpl_normalize.py

```diff
@@ -76,8 +76,16 @@ def __init__(self, data=None, interval=None, vmin=None, vmax=None,
             if self.vmax is None:
                 self.vmax = _vmax
 
+        if stretch is not None and not isinstance(stretch, BaseStretch):
```

Comment:
Just `if not isinstance(stretch, BaseStretch)` is sufficient, no? Even if `stretch` is `None`, it is going to evaluate to `False`.